ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -0,0 +1,331 @@
+     *
+     * @sample maps/demo/geoheatmap/
+     *         GeoHeatmap chart
+     *
+     * @extends      plotOptions.map
+     * @since        next
+     * @product      highmaps
+     * @excluding    allAreas, dragDrop, findNearestPointBy, geometry, joinBy, negativeColor, onPoint,
+     * @requires     modules/geoheatmap
+     * @optionparent plotOptions.geoheatmap
+     */
+
+    public static defaultOptions: GeoHeatmapSeriesOptions =
+        merge(MapSeries.defaultOptions, {
+
+            nullColor: 'transparent',
+
+            tooltip: {
+                pointFormat: 'Lat: {point.lat}, Lon: {point.lon}, Value: {point.value}<br/>'
+            },
+
+            /**
+             * The border width of each geoheatmap tile.
+             *
+             * In styled mode, the border stroke width is given in the
+             * `.highcharts-point` class.
+             *
+             * @type      {number|null}
+             * @default   1
+             * @product   highmaps
+             * @apioption plotOptions.geoheatmap.borderWidth
+             */
+            borderWidth: 1,
+
+            /**
+             * The column size - how many longitude units each column in the
+             * geoheatmap should span.
+             *
+             * @sample maps/demo/geoheatmap/
+             *         GeoHeatmap chart
+             *
+             * @type      {number}
+             * @default   1
+             * @since     next
+             * @product   highmaps
+             * @apioption plotOptions.geoheatmap.colsize
+             */
+            colsize: 1,
+
+            /**
+             * The main color of the series. In heat maps this color is rarely
+             * used, as we mostly use the color to denote the value of each
+             * point. Unless options are set in the [colorAxis](#colorAxis), the
+             * default value is pulled from the [options.colors](#colors) array.
+             *
+             * @type      {Highcharts.ColorString|Highcharts.GradientColorObject|Highcharts.PatternObject}
+             * @since     next
+             * @product   highmaps
+             * @apioption plotOptions.geoheatmap.color
+             */
+
+            /**
+             * The rowsize size - how many latitude units each row in the
+             * geoheatmap should span.
+             *
+             * @sample maps/demo/geoheatmap/
+             *         GeoHeatmap chart
+             *
+             * @type      {number}
+             * @default   1
+             * @since     next
+             * @product   highmaps
+             * @apioption plotOptions.geoheatmap.rowsize
+             */
+            rowsize: 1,
+
+            /**
+             * The z index of the geoheatmap series.
+             *
+             * @type      {number}
+             * @since     next
+             * @product   highmaps
+             * @default   2
+             * @apioption plotOptions.geoheatmap.zIndex
+             */
```

Comment:
Why do we need this? Usually, the z index of the series is determined by its index in the `series` option. The [index](https://api.highcharts.com/highcharts/series.line.index) option lets us override that, and then the [zIndex](https://api.highcharts.com/highcharts/series.line.zIndex) allows us to define a visual z index that differs from the `index`.

Unless the geoheatmap has a special reason for handling this differently, we should just inherit the regular Series behaviour.